thoth/metrics_exporter/jobs/common.py

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env python3
+# thoth-metrics
+# Copyright(C) 2018, 2019 Christoph Görn, Francesco Murdaca, Fridolin Pokorny
+#
+# This program is free software: you can redistribute it and / or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""A collection of methods that can be resued in different metric classes."""
+
+import logging
+
+from datetime import datetime
+from prometheus_api_client import PrometheusConnect
+import thoth.metrics_exporter.metrics as metrics
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def get_workflow_duration(
+    service_name: str,
+    prometheus: PrometheusConnect,
+    namespace: str,
+    check_time: datetime,
+    metric_type: metrics
+) -> datetime:
+    """Get the time spent for each worflow for a certain service."""
+    workflow_status_metric_name = "argo_workflow_status_phase"
+    workflow_status_metrics = prometheus.get_current_metric_value(
+        metric_name=workflow_status_metric_name,
+        label_config={
+            'instance': f"workflow-controller-metrics-{namespace}.cloud.paas.psi.redhat.com:80",
```

Comment:
Maybe parametrize this as well?

Reply:
So not only the namespace but  INSTANCE_URL will be different depending on the service:
- INSTANCE_URL_BACKEND for argo deployed in backend namespace (adviser)
- INSTANCE_URL_MIDDLETIER for argo deployed in middletier namespace (solver)
- INSTANCE_URL_INSPECTION for argo deployed in amun-inspection namespace (inspection)

All three can be added to thoth deployment configmap

Reply:
+1, please be more descriptive - having just "instance" seems to be too generic env var naming